lib/floe/workflow.rb

```diff
@@ -161,13 +161,20 @@ def current_state
     private
 
     def step_next
-      if context.next_state
-        # if rerunning due to an error (and we are using Retry)
-        if context.state_name == context.next_state && context.failed? && context.state.key?("Retrier")
-          restore_values = context.state.slice("RetryCount", "Input", "Retrier")
-        end
-        context.state = {"Name" => context.next_state, "Input" => context.output}.merge!(restore_values || {})
+      context.state_history << context.state
+
+      return unless context.next_state
+
+      next_state = {"Name" => context.next_state, "PreviousStateGuid" => context.state["Guid"]}
```

Comment:
Nice. Introducing `next_state` removes that `restore_values` which was a little confusing.

Reply:
```suggestion
      next_state = {"Name" => context.next_state, "Guid" => SecureRandom.uuid, "PreviousStateGuid" => context.state["Guid"]}.compact
```

and remove: https://github.com/ManageIQ/floe/blob/768c4944a1ab529c823bb29781948969a8df0f88/lib/floe/workflow/state.rb#L56